### PR TITLE
CON-93 Set up dynamic loading of blog posts while scrolling

### PR DESCRIPTION
**Pull request by scott-m-king**
## Context

Add lazy loaded blog posts (20 posts per scroll)

## Describe your changes

- Add `useInfiniteQuery` call for blog posts
- Load 20 posts at a time, when scrolled to the bottom, load 20 more
- Update trpc endpoint to use paging
- Move blog post viewer to `BlogPostViewer` component so that it's reusable for the Profile page

## Issue ticket number and link

https://continuus.atlassian.net/browse/CON-93

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] I verified that my code will compile by running `npm run build` 
- [ ] I have run `npm run lint` and fixed linting errors

### CODE REVIEW

File: src/components/BlogPostViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        (result, curr) => [...result, ...curr.items],
+        [] as Post[]
+      ),
+    [blogPosts]
+  );
+
+  const { ref, inView } = useInView();
+
+  useEffect(() => {
+    console.log(inView);
```
Comment (altarrok) [inline src/components/BlogPostViewer.tsx]: remove log statement pls :)

File: src/components/BlogPostViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import { Spinner } from "./Spinner";
+import { BlogPost as BlogPostComponent } from "./BlogPost";
+
+type Post = BlogPost & { user: User; comments: (Comment & { user: User })[] };
+
+export const BlogPostViewer: React.FC = () => {
+  const [view, setView] = useState<string>("Recent");
+  const { data: blogPosts, fetchNextPage } = api.blogPost.get.useInfiniteQuery(
+    {
+      take: 10,
```
Comment (altarrok) [inline src/components/BlogPostViewer.tsx]: Just curious, does 10 posts fill the page?

File: src/components/BlogPostViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                  comments={comments as (Comment & { user: User })[]}
+                  poster={name as string}
+                  lastUpdated={updatedAt}
+                  post={content}
+                  posterAvatarUrl={image as string}
+                />
+              </>
+            )
+          )}
+      <div className="my-5 flex justify-center" ref={ref}>
```
Comment (altarrok) [inline src/components/BlogPostViewer.tsx]: What happens if there are no more blog posts to fetch?

**Review by altarrok — CHANGES_REQUESTED** [review]
Minor changes and some questions, thank you!!

File: src/components/BlogPostViewer.tsx (hunk 2 above)
Comment (algo-1) [reply src/components/BlogPostViewer.tsx]: Yes it does from testing unless the screen size is really big, perhaps we increase it to 15/20 just to be safe?

File: src/components/BlogPostViewer.tsx (hunk 3 above)
Comment (algo-1) [reply src/components/BlogPostViewer.tsx]: The spinner still shows

Comment (altarrok) [reply src/components/BlogPostViewer.tsx]: Can we not show the spinner when there are no futher posts left to show to the user? (using hasNextPage)

File: src/components/BlogPostViewer.tsx (hunk 2 above)
Comment (altarrok) [reply src/components/BlogPostViewer.tsx]: Its ok, I was just curious

File: src/components/BlogPostViewer.tsx (hunk 3 above)
Comment (scott-m-king) [reply src/components/BlogPostViewer.tsx]: fixed

File: src/components/BlogPostViewer.tsx (hunk 1 above)
Comment (scott-m-king) [reply src/components/BlogPostViewer.tsx]: removed

File: src/components/BlogPostViewer.tsx (hunk 2 above)
Comment (scott-m-king) [reply src/components/BlogPostViewer.tsx]: I increased it to 20 as per the acceptance criteria. 10 does fill up the screen, but executes more queries so I've set it back to 20

**Review by altarrok — APPROVED** [review]

**Pull request merged**